site/website/blog/2020-10-05-jet-43-is-released.md

```diff
@@ -0,0 +1,188 @@
+---
+title: Jet 4.3 is Released
+author: Marko Topolnik
+authorURL: https://twitter.com/mtopolnik
+authorImageURL: https://pbs.twimg.com/profile_images/922726943614783488/Pb5DDGWF_400x400.jpg
+---
+
+Today we're releasing Hazelcast Jet 4.3, our fourth release in 2020!
+
+We took part in Google Summer of Code that ended just a few weeks ago,
+and this release already brings a productionized piece of work that our
+student, [Mohamed Mandouh](https://github.com/MohamedMandouh), made:
+distributed in-memory sorting. Mohamed's primary focus was research into
+the feasibility of integrating RocksDB or a similar DB library as a
+disk-based state backend for Jet's hash join, aggregation and sorting,
+and we plan to continue with this work for some more time.
+
+Here are the main improvements in this release:
+
+## BatchStage.sort()
+
+As mentioned, this is the work coming out of this year's GSoC. You can
+now sort the data coming out of a pipeline stage. For example, this
+starts from an asceding sequence 0..9,999, sorts it in descending order,
+and prints the result:
+
+```java
+var pipeline = Pipeline.create();
+var integerSequence = TestSources.items(IntStream.range(0, 10_000)
+        .boxed().toArray(Integer[]::new));
+pipeline.readFrom(integerSequence)
+        .sort(ComparatorEx.comparing(i -> -i))
```

Comment:
This minus trick to get descending order is an unnecessary embellishment, which will just confuse and not add value. Yes, it's clever, but I would remove it.

Reply:
Removing it means there's no sorting at all, since we started from an ascending sequence. I considered some other options, like adding a randomizing stage, but this was the simplest code to present.